back/helpers.ts

```diff
@@ -0,0 +1,32 @@
+import { getUserOrderCountByDate } from "./database";
+
+const PURCHASE_LIMIT_PER_DAY = 3;
+
+export const canMakeNewPurchase = async (customerDocument: string) => {
+  try {
+    const count = await getUserOrderCountByDate(customerDocument);
+    if (count === undefined || count === null) {
+      return false;
+    }
+    // user has few purchases in total, so it means can make new purchase
+    if (count >= PURCHASE_LIMIT_PER_DAY) {
+      return false;
```

Comment:
Aqui penso que você não precisa validar se count é undefined ou null. Vamos pensar no cenário onde count é null ou undefined:
null ou undefined >= PURCHASE_LIMIT_PER_DAY? é o que você precisa saber.

Reply:
Oii nass! não sei se entendi bem, mas se eu tirar a validação de null ou undefined, se o count for null ou undefined então não cairia na condição if (count >= PURCHASE_LIMIT_PER_DAY), retornando true ao inves do false. Fiz esse exemplo pra ilustrar:
<img width="1385" alt="image" src="https://github.com/luane-aquino/checkout/assets/10707843/a9ee469a-095e-4992-a2c5-1c4c847d8691">